draft-ietf-masque-h3-datagram.md

```diff
@@ -357,6 +357,22 @@ the same order.
 Endpoints which receive a Capsule with an unknown Capsule Type MUST silently
 drop that Capsule.
 
+## Error Handling
+
+When an error occurs processing the capsule protocol, the receiver abruptly
+terminates the enclosing stream.  When running over HTTP/3 or HTTP/2, the
+receiver resets the stream.  When running over HTTP/1.1, the receiver terminates
+the underlying connection with an error.
+
+Each capsule's payload MUST contain exactly the fields identified in its
+description. A capsule payload that contains additional bytes after the
+identified fields or a capsule payload that terminates before the end of the
+identified fields MUST be treated as a stream error of type TBD. In
```

Comment:
Can we define a new CAPSULE_ERROR? That'll be more helpful than TBD

Reply:
I thought about doing that here, but it seemed to violate the thesis.  You want it in this PR?

Reply:
What do you mean by "violate the thesis"? This is a stream error and not a connection error right?

Reply:
Sorry, I meant "the thesis of this PR", which is about how to handle malformed capsules, not creating a new application error code space for capsule protocol.

Reply:
I didn't mean creating a new application error code space, just add a new value to the h3 errors registry

Reply:
I don't think Capsule Protocol errors belong in the H3 space, since Capsule Protocol can be conveyed over other versions of HTTP, right?

Reply:
Capsules travel end-to-end, but HTTP aborts are hop-by-hop. Since we don't have a way to abort end-to-end, we're stuck with using an error from the corresponding HTTP version in use. But you're right, you'd need to allocate one for h2 as well, and mention that the stream abort mechanism in h1 is TCP RST.

Reply:
> and mention that the stream abort mechanism in h1 is TCP RST.

Is this text sufficient?

>  When running over HTTP/1.1, the receiver terminates the underlying connection with an error.

I tried to use the language from core.

Reply:
SGTM. I just rather we either not specify the h3/h2 errors, or define them with a specific value as opposed to TBD.

Reply:
It isn't a problem to use H2/H3 error codes to communicate application (higher-layer) problems. I think it is useful to include the reason (bad capsule) but I could live with `H3_GENERAL_PROTOCOL_ERROR` or the H2 equivalent. 

If we consider a setup where there is an H3 -> H1.1 intermediary, the server behind the intermediary might detect a bad capsule and do a TCP RST. The intermediary can't know the reason for that and would probably end up having to use H3_GENERAL_PROTOCOL_ERROR. So the client might have to handle such a scenario anyway.